android/library/src/main/java/com/polidea/multiplatformbleadapter/ConnectionState.java

```diff
@@ -0,0 +1,12 @@
+package com.polidea.multiplatformbleadapter;
+
+public enum ConnectionState {
+
+    CONNECTING("connecting"), CONNECTED("connected"), DISCONNECTING("disconnecting"), DISCONNECTED("disconnected");
```

Comment:
What about disconnecting event? It is not used. We can consider sending it, for instance, after cancelDeviceConnection invocation.

Reply:
I'm not sure what the native stack report about the connection's state If the connection has been cancelled while being established.
I mean:
1. connected -(cancel)->disconnecting->disconnected - COOL
2. connecting -(cancel)->huh?->disconnected

It should mirror the native behaviour, but @dariuszseweryn said there might be discrepancies between connection state reported by the system and the one by the library, so I'd vote on omitting it altogether and leaving it in code for the sake of consistency.